crates/recorder/src/benchmark.rs

```diff
@@ -119,68 +124,100 @@ impl<'a, 'b> Instance<'a, 'b> {
 pub fn benchmark(
     bench_api: &mut BenchApi,
     wasm_bytes: &[u8],
-    measure_type: MeasureType,
-) -> Result<BenchmarkMeasurements> {
-    let mut measure = measure_type.build();
+    measure: &mut impl Measure,
+    measurements: &mut Measurements,
+) -> Result<()> {
     let engine = Engine::new(bench_api);
 
     // Measure the module compilation.
-    let (module, compilation) = engine.compile(wasm_bytes, &mut measure);
-    info!("Compiled successfully: {:?}", compilation);
+    let module = engine.compile(wasm_bytes, measure, measurements);
+    info!("Compiled successfully");
 
     // Measure the module instantiation.
-    let (instance, instantiation) = module.instantiate(&mut measure);
-    info!("Instantiated successfully: {:?}", instantiation);
+    let instance = module.instantiate(measure, measurements);
+    info!("Instantiated successfully");
```

Comment:
Another thought: I'm adding some more benchmarks and alternating between building them and running to test things out. Having the option to log the measured times is really helpful as a sanity check.